src/js/common/schemaform/FormApp.jsx

```diff
@@ -102,6 +124,27 @@ class FormApp extends React.Component {
     return message;
   }
 
+  redirectOrLoad(props) {
+    const firstPagePath = props.routes[props.routes.length - 1].pageList[0].path;
+
+    // If we're logged in and have a saved / pre-filled form, load that
+    if (props.isLoggedIn) {
+      const currentForm = props.formConfig.formId;
+      const isPrefill = props.prefillsAvailable.includes(currentForm);
+      if (props.savedForms.includes(currentForm) || isPrefill) {
+        props.fetchInProgressForm(currentForm, props.formConfig.migrations, isPrefill);
+      } else {
+        // No forms to load; go to the beginning
+        // If the first page is not the intro and uses `depends`, this will probably break
+        props.router.push(firstPagePath);
```

Comment:
What do you think about using replace instead of push here (and the other places)? If they click back after being redirected it's going to be surprising, since they won't have seen the page they're going back to.